components/Contributors/AddToContributorInsightDrawer.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { BsGithub } from "react-icons/bs";
+import { usePostHog } from "posthog-js/react";
+import SingleSelect from "components/atoms/Select/single-select";
+import Button from "components/shared/Button/button";
+import { fetchApiData } from "helpers/fetchApiData";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+import { useToast } from "lib/hooks/useToast";
+import Text from "components/atoms/Typography/text";
+import { useFetchAllLists } from "lib/hooks/useList";
+import useWorkspaces from "lib/hooks/api/useWorkspaces";
+import { Drawer } from "components/shared/Drawer";
+
+type AddToContributorInsightDrawerProps = {
+  repository: string;
+  contributors: string[];
+};
+
+export default function AddToContributorInsightDrawer({
+  repository,
+  contributors,
+}: AddToContributorInsightDrawerProps) {
+  const { toast } = useToast();
+  const router = useRouter();
+  const posthog = usePostHog();
+  const { signIn, user, sessionToken } = useSupabaseAuth();
+  const { data: contributorInsights, isLoading } = useFetchAllLists();
+  const [workspaceId, setWorkspaceId] = useState("new");
+  const { data: workspaces, isLoading: workspacesLoading, mutate } = useWorkspaces({ load: !!user, limit: 100 });
+
+  const [selectedInsight, setSelectedInsight] = useState("new");
+
+  const [host, setHost] = useState("");
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHost(window.location.origin as string);
+    }
+  }, []);
+
+  const addToContributorInsight = async () => {
+    posthog.capture(`Repo Pages: add to contributor insight`, {
+      repository,
+      count: contributors.length,
+    });
+
+    const { error } = await fetchApiData({
+      method: "POST",
+      path: `lists/${selectedInsight}/contributors`,
+      body: {
+        contributors: contributors.map((contributor) => {
+          return { login: contributor };
+        }),
+      },
+      bearerToken: sessionToken!,
+      pathValidator: () => true,
```

Comment:
Refactor (blocking): ensure the `selectedInsight` is a valid UUID via Valibot's `safeParse`.

```suggestion
      pathValidator: () => safeParse(UuidSchema, selectedInsight).success,
```